apps/base-docs/tutorials/docs/0_gassless-transactions-with-paymaster.md

```diff
@@ -0,0 +1,616 @@
+
+![cdp-project-selection.png](../../assets/images/gasless-transaction-on-base/cdp-select-project.png)
+
+**Allowlist the NFT contract and the mintTo functions:**
+
+Click the `Add` button to add a contract.
+
+Add the following contract: `0x83bd615eb93eE1336acA53e185b03B54fF4A17e8`
+
+The function to enable is `mintTo`
+
+![cdp-allowlist-contracts.png](../../assets/images/gasless-transaction-on-base/cdp-allowlist-contract.png)
+
+:::note Use your own contract
+We will be using this simple NFT deployed at X for our example. Feel free to use a contract of your choice.
+:::
+
+**Global Limits:**
+
+This setting allows you to define the maximum amount of gas or USD that can be sponsored globally across all users. It helps control the total expenditure on gas sponsorship, ensuring that the allocated budget is not exceeded. For example, setting a global limit of $1 means that the Paymaster will sponsor transactions until the total gas cost reaches $1.
+
+Set your global policy to be `$.07` by entering the amount in the text field and clicking the `Save` button.
+
+**Scroll down to the “Per User Limit” section:**
+
+We can set a Paymaster policy that specifies either a specific dollar amount or a number of UserOperations. We can enable a 'limit cycle' that allows this policy to refresh every week. This feature enables applications to sponsor smart wallets on a weekly, daily, or monthly basis, either by amount or by number of UserOperations.
+
+Set the max USD to $0.05 and the max UserOperation to 1 to create a policy with a maximum sponsorship limit of $0.05 and a maximum of 1 UserOperation per user.
+
+![cdp-allowlist-contracts.png](../../assets/images/gasless-transaction-on-base/cdp-global-user-limits.png)
+
+### Test your Paymaster policy
+
+We will now test the policy that was created.
+
+:::info The Use of Foundry
+
+In this tutorial, we use foundry to create two key pairs that will allow us to create Smart Accounts that we will sponsor transactions on behalf of. You will **not** need to send any funds (ETH) to these wallets. You may use private keys of wallets you own as an alternative.
+
+Foundry depends on Rust to work. If you do not have rust install by running `curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh`
+
+For more information, see the [Foundry Book installation guide](https://book.getfoundry.sh/getting-started/installation).
+
+:::
+
+**Open a terminal and install Foundry by running:**
+
+```bash
+curl -L https://foundry.paradigm.xyz | bash
+```
+
+**Install the Foundry toolchain installer:**
+
+```bash
+foundryup
+```
+
+**Create a directory named `sponosored_transactions`:**
+
+```bash
+mkdir sponosored_transactions
+```
+
+**Change into the new directory, initialize a node project, install two dependencies `viem` and `permissionless` and create a file titled `index.js`:**
+
+```bash
+cd paymaster-tutorial
```

Comment:
Is this another folder, or did you mean `sponsored_transactions`?

Reply:
good catch - i def meant `sponsored_transactions`